merlin-framework/src/main/java/gov/nasa/jpl/aerie/merlin/framework/ModelActions.java

```diff
@@ -12,18 +13,38 @@ protected ModelActions() {}
   static final Scoped<Context> context = Scoped.create();
 
 
+  public static <T> Context.TaskFactory threaded(final Supplier<T> task) {
+    return executor -> new ThreadedTask<>(executor, ModelActions.context, task);
+  }
+
   public static Context.TaskFactory threaded(final Runnable task) {
-    return executor -> new ThreadedTask(executor, ModelActions.context, task);
+    return threaded(supplierOfRunnable(task));
+  }
+
+  public static <T> Context.TaskFactory replaying(final Supplier<T> task) {
+    return executor -> new ReplayingTask<>(executor, ModelActions.context, task);
   }
+
   public static Context.TaskFactory replaying(final Runnable task) {
-    return executor -> new ReplayingTask(executor, ModelActions.context, task);
+    return replaying(supplierOfRunnable(task));
   }
 
 
-  public static String spawn(final Runnable task) {
+  public static <T> String spawn(final Supplier<T> task) {
     return spawn(threaded(task));
   }
 
+  public static String spawn(final Runnable task) {
+    return spawn(supplierOfRunnable(task));
+  }
+
+  private static Supplier<VoidEnum> supplierOfRunnable(final Runnable task) {
+    return () -> {
+      task.run();
+      return VoidEnum.VOID;
+    };
+  }
```

Comment:
Personally, I'd be in favor of inlining this method at its call sites. I find `() -> { task.run(); return VoidEnum.VOID; }` more informative than `supplierOfRunnable`, without being *too* verbose.